experiment/measurer/experiment_coverage_utils.py

```diff
@@ -0,0 +1,189 @@
+            name_entries.append([self.function_names[name], name, "function"])
+        for name in self.file_names:
+            name_entries.append([self.file_names[name], name, "file"])
+        self.name_df = pd.DataFrame(name_entries,
+                                    columns=["id", "name", "type"])
+
+    def name_to_id(self, name):  # pylint: disable=no-self-use
+        """Generates a hash for the name. This is to save disk storage"""
+        return hashlib.md5(name.encode()).hexdigest()[:7]
+
+    def remove_redundant_duplicates(self):
+        """Removes redundant entries in segment_df. Before calling this
+        function, for each time stamp, segment_df contains all segments that are
+        covered in this time stamp. After calling this function, for each time
+        stamp, segment_df only contains segments that have been covered since
+        the previous time stamp. This significantly reduces the size of the
+        resulting CSV file."""
+        try:
+            # Drop duplicates but with different timestamps in segment data.
+            self.segment_df = self.segment_df.sort_values(by=['time'])
+            self.segment_df = self.segment_df.drop_duplicates(
+                subset=self.segment_df.columns.difference(['time']),
+                keep="first")
+            self.name_df = self.name_df.drop_duplicates(keep="first")
+
+        except (ValueError, KeyError, IndexError):
+            coverage_utils.logger.error(
+                'Error occurred when removing duplicates.')
+
+    def generate_csv_files(self):
+        """Generates three compressed CSV files containing coverage information
+        for all fuzzers, benchmarks, and trials. To maintain a small file size,
+        all strings, such as file and function names, are referenced by id and
+        resolved in 'names.csv'."""
+
+        # Clean and prune experiment-specific data frames.
+        self.remove_redundant_duplicates()
+
+        # Write CSV files to file store.
+        def csv_filestore_helper(file_name, df):
+            """Helper function for storing csv files in filestore."""
+            src = os.path.join(coverage_utils.get_coverage_info_dir(), 'data',
+                               file_name)
+            dst = exp_path.filestore(src)
+            df.to_csv(src, index=False, compression='infer')
+            filestore_utils.cp(src, dst)
+
+        csv_filestore_helper('functions.csv.gz', self.function_df)
+        csv_filestore_helper('segments.csv.gz', self.segment_df)
+        csv_filestore_helper('names.csv.gz', self.name_df)
+
+
+def extract_segments_and_functions_from_summary_json(  # pylint: disable=too-many-locals
+        summary_json_file, benchmark, fuzzer, trial_id, time):
+    """Return a trial-specific data frame container with segment and function
+     coverage information given a trial-specific coverage summary json file."""
+
+    process_specific_df_container = DataFrameContainer()
```

Comment:
Are you calling this "process_specific" because there will be one for each item we map using the multiprocessing pool?
Because I think that's a weird detail to name this var after (I think it's also wrong, the processes in the pool are reused). I believe we are mapping over trials, can you name at after that (e.g. `trial_df_container`).